src/transformers/models/chameleon/modeling_chameleon.py

```diff
@@ -1343,6 +1343,12 @@ def forward(
 
         if pixel_values is not None:
             image_tokens = self.get_image_tokens(pixel_values)
+            n_image_tokens_in_text = (input_ids == self.vocabulary_mapping.image_token_id).sum().item()
```

Comment:
Mmmm in general I don't mind, as this should help our users, but the .item() might break compile compatibility (well only full graph). 

@McPatate that's where and when we would need to see how much we are losing from this small change ! 🤗 (FYI @LysandreJik )

Reply:
we can alwayss wrap these in `is_torchdynamo_compiling`, same was a s we wrap all warnings/logging now in generation code. So we ask users to make sure the code works w/o compilation, to see all warning etc, and then compile the code which will not show the exact reason why/where this CUDA-side error was triggered

Reply:
The thing is these `is_compiling` are unrelated to normal users ~-> expose them to unrelated codes

Reply:
i see what you mean. Yes, the processing should maybe check this, but we cannot perform any checks before getting image hidden states. My main idea was to bring the same check we had earlier in `merge_inputs` method, so that after moving to the new logic we still can trace down bugs related to shape mismatch easily, or let users track that down

Also we won't do the sum() and item() every forward, for generation it is only for prefill stage after which we'll have image states in the cache. But anyway, if you think this is too many checks (given we now support old and new logic in VLMs for a few minor releases), I am okay with not adding it. I don't see it as a major blocker or anything, more like a nice addition for users :D

Reply:
Okay let's add it then 🤗